cmd/rpcdaemon/filters/filters.go

```diff
@@ -524,3 +551,78 @@ func generateSubscriptionID() SubscriptionID {
 
 	return SubscriptionID(fmt.Sprintf("%x", id))
 }
+
+func (ff *Filters) AddLogs(id LogsSubID, logs *types.Log) {
+	ff.storeMu.Lock()
+	defer ff.storeMu.Unlock()
+	st, ok := ff.logsStores[id]
+	if !ok {
+		st = make([]*types.Log, 0)
+	}
+	st = append(st, logs)
+	ff.logsStores[id] = st
+}
+
+func (ff *Filters) ReadLogs(id LogsSubID) ([]*types.Log, bool) {
+	ff.storeMu.Lock()
+	defer ff.storeMu.Unlock()
+	res := make([]*types.Log, 0)
+	st, ok := ff.logsStores[id]
+	if !ok {
+		return res, false
+	}
+	res = append(res, st...)
+	st = make([]*types.Log, 0)
```

Comment:
Need add something to st?

Reply:
No, it's just for erasing slice from memory.